### PR TITLE
Test for no EOC connection allowing external I2C baro on acro boards

### PR DESCRIPTION
**Pull request by SteveAmor**
with some baro code and comments cleanup.
Fixes issue #1017

### CODE REVIEW

**Comment by SteveAmor** [comment]
I have tested on an acro Naze but can only simulate the operation on a Full Naze to make sure Full Naze users aren't affected by this code change.

**Comment by SteveAmor** [comment]
Squashed the two commits.

File: src/main/drivers/barometer_bmp085.c [hunk 1]
```diff
@@ -371,4 +374,27 @@ static void bmp085_get_cal_param(void)
...
+bool bmp085TestEOCConnected(const bmp085Config_t *config)
+{
+    uint32_t baroDeadline = 0;
+    uint32_t currentTime = micros();
+
+    baroDeadline += (UT_DELAY * 2); // wait twice as long as normal, just to be sure
+    bmp085_start_ut();
+    baroDeadline += micros();
+
+    while ((int32_t)(currentTime - baroDeadline) < 0) {
```
Comment (ledvinap) [inline src/main/drivers/barometer_bmp085.c]: using `delay(UT_DELAY / 1000 + 1 )` should be OK here. UT_DELAY has margin included.

**Comment by ledvinap** [comment]
Setting input to PULLUP will do too .. 

Also when checking is EOC is connected, full sequence should be:
- check that EOC is high, wait UP_DELAY if not (conversion may be in progress), check again
- start T conversion, delay a bit, check that EOC is low
- wait  UT_DELAY
- check that EOC is high

This is not important in current code (just setting pullup on pin will fix the problem), but it will help if code starts using EOC in future ...

**Comment by SteveAmor** [comment]
@ledvinap Many thanks for the feedback, it's very much appreciated.
I doubled the wait time as the code to use EOC was added to protect against conversion times that exceeded the datasheet max and as this code is only executed, once, during initialisation, I wasn't too concerned.
I like the simplicity of my test and will ensure it's only possible to do during initialisation so there would not be a conversion in process when called.  My checking code has to use a pulldown as EOC goes high at the end of the conversion.
I do like the resiliance of your code but there is nothing in the datasheet that would suggest a time to wait after you start T conversion and check the EOC is low before the conversion is complete (for which there is only a maximum conversion time).

**Comment by stronnag** [comment]
I tried this patch with a BMP180 (BMP085 bug for bug compatible). As without the patch, altitude was not recorded correctly, either in CF Configurator or mwp. From the MSP, vario would change, but estalt always decayed to 0.

It appears (and not because of this patch) that I2C baros are not detected correctly. git master + PR 1033 on an acro baord (flip 32) with a BMP 180 (actually a GY-652 BMP180+HMC5983) reports the following:
```
# status
CPU Clock=72MHz, GYRO=MPU6050, ACC=MPU6050.n, BARO=MS5611, MAG=HMC5883
```
Note the BARO entry. If I remove ms5611 from NAZE/target.h, then the I2C BMP180 works properly on the acro board via I2C.
```
# status
CPU Clock=72MHz, GYRO=MPU6050, ACC=MPU6050.n, BARO=BMP085, MAG=HMC5883
```

**Comment by SteveAmor** [comment]
@stronnag does it work correctly with v1.8.1?

**Comment by SteveAmor** [comment]
@stronnag 

I have a BMP180 on a GY-68 board.  I removed the I2C pullup resistors on the GY-68 board since they are already on the FC and I power it with 5v as the GY-68 board has a 5v to 3.3v regulator on it.
My FC is an acro Naze.  Here is the status:
# status
System Uptime: 6 seconds, Voltage: 0 * 0.1V (1S battery)
CPU Clock=72MHz, GYRO=MPU6050, ACC=MPU6050.n, BARO=BMP085
Cycle Time: 3551, I2C Errors: 0, config size: 1724

Also: Board: AFNA, version: 2

Have you removed the I2C pullup resistors on your board?  The code checks for an MS5611 first but should then check for a BMP085 (or drop in replacement equivilent).

**Comment by stronnag** [comment]
@SteveAmor 

The Mag works as expected. The GY-652 also has a 5V reg (so powered with 5v). I did not remove any resistors.
Board: AFNA, version: 1
I'll put 1.8.1 on it later.

**Comment by SteveAmor** [comment]
@stronnag I assume the baro detection doesn't work properly in 1.9.0 without PR #1033.  Be good to understand if baro detection is also a problem in 1.8.1. At least there is a small difference in our board detection version that may be the cause.

**Comment by stronnag** [comment]
@SteveAmor 

I disabled the pull ups (power off 3.3v). This has the useful side effect of allowing the board to boot when powered off USB with the GY-652 connected. The Baro is still detected as a non-existing MS5611.

If I flash 1.81 (stable release), the baro is also mis-detected as a MS5611.

**Comment by SteveAmor** [comment]
@stronnag: I have raised the issue of your BMP180 being falsely detected as an MS5611 under issue #1040

**Comment by skaman82** [comment]
Thanks! Just tested this. It works now for me :) I have the acro naze (Board: AFNA, version: 2) and the GY-68-BMP180 connected (SDA/SDL + 3v). Will try it with GY-63 MS5611-01BA03 in the next days.

**Comment by fusterjj** [comment]
This fixes the issue for me as well.  External BMP180 + Acro Naze32.

**Comment by aequitas** [comment]
Work for me to, naze32 acro, BMP180. Have not tried tuning, but baro readout in sensor tab shows better output then before.

**Comment by skaman82** [comment]
Tested with the GY-63 breakout board. The MS5611 baro is recognized and show data on the sensors tab. Although the data shown in the sensors tab is not very precise. The altitude slowly moves +-1m without moving the baro. But I'm hoping @hydra will merge this and it makes in CF 1.9.1! Also tested with GY-68-BMP180. Works OK.

**Comment by canphaz** [comment]
Works on Vortex, just tried it. Readings are visible in Cleanflight Sensors tab.

**Comment by skaman82** [comment]
Does this makes into CF 1.10?

**Comment by stronnag** [comment]
@hydra, for 1.10 please

**Comment by skaman82** [comment]
@hydra, for 1.10 please. Very important bugfix for some people. Tested this today in flight. Works very well. Please merge in to master. Thx

**Comment by skaman82** [comment]
yay!

**Comment by digitalentity** [comment]
@hydra thanks alot!!!

**Pull request merged**